src/routes.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import Dashboard from './containers/Dashboard'
+import {Login} from './containers/Login'
+import Profile from './containers/Profile'
+import { AuthStore } from './repos/UserRepo'
+import {
+Route,
+Switch,
+Redirect 
+} from 'react-router-dom'
+
+
+const  renderIfAuthenticated = (Component) => props => AuthStore.isAuthenticated ? 
+(<Component {...props} />) : 
+(<Redirect
+    to={{
+      pathname: "/login",
+      state: { from: props.location }
+    }}
+  />
+)
+
+const PrivateRoute = ({ component: Component, ...rest }) => (
+<Route
+  {...rest}
+  render={renderIfAuthenticated(Component)}
+/>
+);
+
+
+export const routerOutput = () =>
+    (
+    <Switch>
```

Comment:
`Switch` will only render the first matching route (from top to bottom). 

because without the switch router by default will render all matching routes.
if the first route didn't have exact it will be matched with every url and you will get stuck in a redirect loop.

one addition can be is a catch-all route to match any incorrect urls (at the end of all routes) : 
`<Route component={NotFoundComponent} />`